cmd/jaeger/internal/extension/remotesampling/config.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package remotesampling
+
+import (
+	"errors"
+	"reflect"
+	"time"
+)
+
+var (
+	errNoSource       = errors.New("no sampling strategy specified, has to be either 'adaptive' or 'static'")
+	errMultipleSource = errors.New("only one sampling strategy can be specified, has to be either 'adaptive' or 'static'")
+)
+
+type StaticConfig struct {
+	// File specifies a local file as the strategies source
+	File string `mapstructure:"file"`
+
+	// ReloadInterval determines the periodicity to refresh the strategies
+	ReloadInterval time.Duration `mapstructure:"reload_interval"`
+}
+
+type AdaptiveConfig struct {
```

Comment:
A lot of the parameters here are actually only needed for the adaptive sampling processor, which should run inside span processor, not inside the extension.

Reply:
I tried doing it, but it makes the code more complicated. We call `NewStrategyStore` in the extension, which requires all options. However, in the adaptive sampling processor, we only need these options: `AggregationBuckets` and `CalculationInterval` to create a new aggregator that calculates probability and stores it.

But if we still need to pass this configuration to the processor, we must find a way to transport the configuration from processor to the remote sampling extension internally.

Reply:
Currently, I cannot find any way to transport the configuration internally from the processor to the extension.

cc @james-ryans

Reply:
I was wrong, most params are indeed used by `adaptive.Processor`, which is confusing since it's very different in purpose from the new `adaptivesampling` processor that you added. I think it's actually a poor design choice in the old code, because what that `Processor` does is it periodically loads the throughput numbers, calculates new probabilities and stores them. Then the "strategy store" part, which is really needed for the HTTP endpoint, only reads the probabilities and constructs sampling strategies. We can see the evidence here - starting two goroutines at once is a smell, breaking single responsibility principle: https://github.com/jaegertracing/jaeger/blob/5ddcaa98811e34f7b1292ceb8fe0ff781640b678/plugin/sampling/strategystore/adaptive/processor.go#L174-L175

Arguably a better design would've been for the Processor part to be co-located with the Aggregator (and run by `adaptivesamping` processor), and only retrieval/serving logic would've been in the `remotesampling` extension.

Think about it. Maybe we can do a bit of preemptive refactoring of the `adaptive.Processor` class:
  * keep the part that runs in the background and updates storage with probabilities
  * move the loadProbabilities/runUpdateProbabilitiesLoop part into strategy_store.go (as a separate class).

But one thing to be careful about is if we do this we want to ensure that these config options are cleanly split into two disjoined sets. Otherwise users would have to match the options across two OTEL components, something I would prefer to avoid. Right now they only share the storage name (if you follow my suggestion in another comment about `maxBuckets`).

Reply:
Ok, I'm creating a new PR to refactor `v1` sampling.